sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/kafka/AzureEventHubsKafkaOAUTH2Configuration.java

```diff
@@ -0,0 +1,108 @@
+    private final TokenCredential defaultTokenCredential;
+
+    private static final String SPRING_BOOT_KAFKA_PROPERTIES_BEAN_NAME = "spring.kafka-org.springframework.boot"
+        + ".autoconfigure.kafka.KafkaProperties";
+
+    AzureEventHubsKafkaOAUTH2Configuration(@Qualifier(SPRING_BOOT_KAFKA_PROPERTIES_BEAN_NAME) KafkaProperties kafkaProperties,
+                                           AzureTokenCredentialResolver resolver,
+                                           @Qualifier(DEFAULT_TOKEN_CREDENTIAL_BEAN_NAME) TokenCredential defaultTokenCredential,
+                                           AzureGlobalProperties azureGlobalProperties) {
+        this.kafkaProperties = kafkaProperties;
+        this.tokenCredentialResolver = resolver;
+        this.defaultTokenCredential = defaultTokenCredential;
+        this.azureGlobalProperties = azureGlobalProperties;
+    }
+
+    @Bean
+    DefaultKafkaConsumerFactoryCustomizer defaultKafkaConsumerFactoryCustomizer() {
+        Map<String, Object> updateConfigs = new HashMap<>();
+        Map<String, Object> consumerProperties = kafkaProperties.buildConsumerProperties();
+        if (needConfigureSaslOAuth(consumerProperties)) {
+            AzureThirdPartyServiceProperties azureKafkaConsumerProperties = buildAzureProperties(consumerProperties,
+                azureGlobalProperties);
+            updateConfigs.put(AZURE_TOKEN_CREDENTIAL, buildCredentialMapFromProperties(azureKafkaConsumerProperties));
+            updateConfigs.putAll(KAFKA_OAUTH_CONFIGS);
+        }
+        return factory -> factory.updateConfigs(updateConfigs);
+    }
+
+    @Bean
+    DefaultKafkaProducerFactoryCustomizer defaultKafkaProducerFactoryCustomizer() {
+        Map<String, Object> updateConfigs = new HashMap<>();
+        Map<String, Object> producerProperties = kafkaProperties.buildProducerProperties();
+        if (needConfigureSaslOAuth(producerProperties)) {
+            AzureThirdPartyServiceProperties azureKafkaProducerProperties = buildAzureProperties(producerProperties,
+                azureGlobalProperties);
+            updateConfigs.put(AZURE_TOKEN_CREDENTIAL, buildCredentialMapFromProperties(azureKafkaProducerProperties));
+            updateConfigs.putAll(KAFKA_OAUTH_CONFIGS);
+        }
+        return factory -> factory.updateConfigs(updateConfigs);
+    }
+
+    private TokenCredential buildCredentialMapFromProperties(AzureThirdPartyServiceProperties azureKafkaConsumerProperties) {
+        TokenCredential tokenCredential = tokenCredentialResolver.resolve(azureKafkaConsumerProperties);
+        if (tokenCredential == null) {
+            tokenCredential = defaultTokenCredential;
+        }
+        return tokenCredential;
+    }
+
+    @ConditionalOnClass(KafkaMessageChannelBinder.class)
+    @Configuration(proxyBeanMethods = false)
+    @Import(KafkaBinderConfiguration.class)
+    static class AzureKafkaSpringCloudStreamAutoConfiguration {
```

Comment:
This is not a AutoConfiguration class, so maybe call it `AzureKafkaSpringCloudStreamConfiguration`